### DISCUSSION TITLE
Question on roles and permissions

### DISCUSSION THREAD

**Discussion by heidivanparys** [ec-jrc/re3gistry#334] [Q&A]
I am trying to understand the following lines of code and the rationale behind it:

https://github.com/ec-jrc/re3gistry/blob/82ace9d481d3ce72ea4e0bd489698228a2218a51/sources/Re3gistry2/src/main/resources/configurations_files/configuration.properties#L137-L143

I made the following mapping, to get a better overview:

<table>
    <tr>
        <th></th>
        <th>control body</th>
        <th>register manager</th>
        <th>register owner</th>
        <th>registry manager</th>
        <th>submitting organisation</th>
    </tr>
    <tr>
        <th colspan="6">Content editing</th>
    </tr>
    <tr>
        <th>submit proposal</th>
        <td></td>
        <td></td>
        <td></td>
        <td>(x)<sup>1</sup></td>
        <td>x</td>
    </tr>
    <tr>
        <th>manage item proposal</th>
        <td></td>
        <td></td>
        <td>(x)<sup>2</sup></td>
        <td>x</td>
        <td>x</td>
    </tr>
    <tr>
        <th>approve proposal</th>
        <td>x</td>
        <td></td>
        <td></td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <th>publish proposal</th>
        <td></td>
        <td>x</td>
        <td></td>
        <td></td>
        <td></td>
    </tr>
    <tr>
        <th colspan="6">User &amp; group management</th>
    </tr>
    <tr>
        <th>manage group</th>
        <td></td>
        <td></td>
        <td>x</td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th>manage user</th>
        <td></td>
        <td></td>
        <td>x</td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th>map user to group</th>
        <td></td>
        <td></td>
        <td>x</td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th>map group to item</th>
        <td></td>
        <td></td>
        <td></td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th colspan="6">Structure &amp; data model management</th>
    </tr>
    <tr>
        <th>manage field</th>
        <td></td>
        <td></td>
        <td></td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th>manage field mapping</th>
        <td></td>
        <td>x</td>
        <td></td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th>manage register &amp; registry</th>
        <td></td>
        <td></td>
        <td></td>
        <td>x</td>
        <td></td>
    </tr>
    <tr>
        <th colspan="6">System management</th>
    </tr>
    <tr>
        <th>manage system</th>
        <td></td>
        <td></td>
        <td></td>
        <td>x</td>
        <td></td>
    </tr>
</table>

<sup>1</sup> Present in the default configuration, not present in the line that is commented out.
<sup>2</sup> Not present in the default configuration, present in the line that is commented out.

My questions:

1. What is the reason behind the lines that are commented out? Was the configuration different earlier? Why was it changed?
2. What does it mean exactly to map groups to items? I read the documentation, but I can't seem to find anything about that.
3. Are "manage group", "manage user", "manage user to group" and "map group to item" still relevant, or are they perhaps part of another permission? I'm asking because I tried to figure out what exactly `MapGroupToItem` meant, and in that process I noticed that I got no results back when searching the code for references regarding these four permissions, see also screenshot below for what it looks like for one of them. Am I missing something?

![image](https://user-images.githubusercontent.com/11427611/132238147-c7b202e4-9dba-4d8a-b885-8975d6e031eb.png)

**Comment by heidivanparys** [comment]
Additional question: the comment at 

https://github.com/ec-jrc/re3gistry/blob/6bf4e416d16fbc4d3b98573d07ddaef39022a7f6/documentation/user-manual.md#L475

does not seem to correspond with

https://github.com/ec-jrc/re3gistry/blob/82ace9d481d3ce72ea4e0bd489698228a2218a51/sources/Re3gistry2/src/main/resources/configurations_files/configuration.properties#L137-L140

Can a registry manager (that does not have the "register manager" role) publish reference codes, or not?

**Comment by emanuelaepure10** [comment]
Dear @heidivanparys 

The re3gistry/documentation/user-manual.md is correct.
Both REGISTER and REGISTRY managers have the role of PUBLISHING a change/action.


The variables from the file re3gistry/sources/Re3gistry2/src/main/resources/configurations_files/configuration.properties
are not used, but they could actually give an idea of what a user with a role can do. 
They can be updated, so as well the registryManager has the role of PublishProposal.

Please, be free to update the file.

Best regards,
Emanuela
